cosmwasm/packages/wasmi-runtime/src/recursion_depth.rs

```diff
@@ -0,0 +1,34 @@
+use std::sync::SgxMutex;
+
+use lazy_static::lazy_static;
+
+use enclave_ffi_types::EnclaveError;
+
+const RECURSION_LIMIT: i32 = 5;
```

Comment:
Why not `u32`?

Reply:
No reason, really. It's the default integer type. our limit is so low that this can be any integer type

Reply:
Sorry I meant to comment this about `RECURSION_DEPTH`.
I'm trying to think whether an attacker can screw with `RECURSION_DEPTH` and somehow get to a negative count.

I'd turn them both to `u8` and use `saturating_add` and `saturating_sub`.

Also, let's assume that somehow queries and txs work concurrently, E.g. handle called a sub-query, and while inside the `ocall` the program decides to switch to a query thread. Now two different contexts share the same counter.
The counter protects against 5 (6?) VMs in parallel, but now the tx might be cut short because of a local query and this will kick the node out of consensus.
Do we want to protect against that as well, or are we good with this?

Reply:
I would like to protect against this, but that would require thread local variables (which we should definitely try to tackle again).
In the meantime though, the enclave only has one TCS so only one thread can interact with it at a time (I think?)